### ISSUE TITLE
The input_dim of convolution2d layer

### ISSUE THREAD

**Issue by zherongz** [fchollet/keras#3385]
I try to construct a DNN, but meet a problem.Here it is 
Exception: Input 0 is incompatible with layer convolution2d_2: expected ndim=4, found ndim=3

model = Sequential()
model.add(Reshape((57, 47), input_shape=(2679,)))
model.add(Convolution2D(10,5,5,border_mode='same'))
model.add(MaxPooling2D(pool_size=(2, 2), strides=None, border_mode='valid', dim_ordering='th'))
model.add(Dropout(0.5))
model.add(Convolution2D(15,5, 5, border_mode='same'))
model.add(MaxPooling2D(pool_size=(2, 2), strides=None, border_mode='valid', dim_ordering='th'))
model.add(Dropout(0.5))
model.add(Flatten())
model.add(Dense(output_dim = 2000))
model.add(Activation('tanh'))
model.add(Dense(40, init='uniform'))
model.add(Activation('softmax'))
model.compile(optimizer = 'rmsprop', loss = 'categorical_crossentropy', metrics=['accuracy'], sample_weight_mode=None)

Any one knows the answer? Tks

**Comment by bernardohenz** [comment]
On the first layer use ``` model.add(Reshape((1,57, 47), input_shape=(2679,))) ```
As far as I know, the Convolution2D expects the input as shape ```(samples,channels,width,height)```
